### PR TITLE
feat: Add MovingAI pathing for NPCs without combatAI

### LINKED ISSUE #555: BUG: Heimlich Stewblaster stands still
### Make sure you've done the following:

- [X] I have checked that this issue has not already been reported.

- [X] I have validated that this issue is not a syntax error of either MySQL or SQLite.

- [X] I have pulled the latest version of the main branch of DarkflameServer and have confirmed that the issue exists there.

### DarkflameServer Version

a1a3453

### Issue Description

The chef vendor in Crux Prime by Sentinel Point Zeta is acting like a typical vendor

### Reproduction Steps

he is found in a quiet spot left of Sentinel Point Zeta

### Expected Behavior

https://youtu.be/_R0-uQ_P_GI?t=2906

He is supposed to waltz back and forth in a small area until you interact with him, where he stops and takes off his backpack

### Environment

ubuntu server

### PR DESCRIPTION
**Pull request by EmosewaMC**
wow look we finally got it done, lets see how old this issue is
image
yeah thats pretty old

fixes #555 

also fixes and unreported issue that apparently we were all just complacent with; enemies no longer do small teleports do go from location to location.  They now follow a smooth path of their actual position which makes combat way more fun.  The point of the Approximate location is kind of defeated now given the position needs to be set on every frame, but combat and moving proximity boxes are now correct.

Tested (in an unmodified vanilla client) that all enemies in crux move in smooth lines towards the player with no teleporting, Forbidden valley ronins and horsemen path smoothly
Battle of Nimbus Station enemies path smoothly
Avant Gardens Survival enemies path smoothly
Heimlich Stewblaster now moves along his path and stops when players are close
Robots in Robot City now move and stop when players interact with them (animation isnt consistent, but a non-issue)
both of the above also loop as per their pathing configuration.

Note: Pathing is only implemented right now for NPCs without CombatAI.  This is because this was a necessary step to get the CombatAI working and a good stepping stone to implementing pathing overrides for CombatAI (see patrols for skeleton commanders in Crux Prime).  

here is a video of the difference as well.
https://www.youtube.com/watch?v=7ZMOTrO34Sg

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,7 +5,7 @@ include(CTest)
 set(CMAKE_CXX_STANDARD 20)
 set(CXX_STANDARD_REQUIRED ON)
 set(CMAKE_POLICY_DEFAULT_CMP0063 NEW) # Set CMAKE visibility policy to NEW on project and subprojects
-set(CMAKE_CXX_VISIBILITY_PRESET hidden) # Set C++ symbol visibility to default to hidden
+# set(CMAKE_CXX_VISIBILITY_PRESET hidden) # Set C++ symbol visibility to default to hidden
```
Comment (jadebenn) [inline CMakeLists.txt:8]: Why is this commented out? If this is affecting compilation, it's likely to be a cross-platform issue given MSVC's default symbol visibility.

Comment (EmosewaMC) [reply CMakeLists.txt:8]: These two lines are removed in another PR.

Comment (jadebenn) [reply CMakeLists.txt:8]: This should not be part of this PR then.

Comment (EmosewaMC) [reply CMakeLists.txt]: Reverted

**Review by aronwk-aaron — APPROVED** [review]

**Review by jadebenn — APPROVED** [review]
Looks good to me!

**Pull request merged**